pkg-manager/core/src/link/index.ts

```diff
@@ -131,7 +131,10 @@ export async function link (
   if (opts.targetDependenciesField) {
     newPkg = await updateProjectManifestObject(opts.dir, opts.manifest, specsToUpsert)
     for (const { alias } of specsToUpsert) {
-      updatedWantedLockfile.importers[importerId].specifiers[alias] = getSpecFromPackageManifest(newPkg, alias)
+      const spec = getSpecFromPackageManifest(newPkg, alias)
+      if (spec !== undefined) {
```

Comment:
You can probably use this just to cover both null and undefined
```suggestion
      if (spec != null) {
```

Reply:
Is it possible for this to have `null`? My reading of the function says that it isn't possible.

Reply:
not possible but do we need to check strictly for undefined only?

Reply:
Yes, because plain truthiness will then skip over versions set to the empty string, changing that behavior (that was the failure that this PR previously had in one of the existing tests).

Reply:
I am not suggesting truthiness

```
❯ node
Welcome to Node.js v16.20.0.
Type ".help" for more information.
> var a = ''
undefined
> a != null
true
>

```

Reply:
Right, but if the value can only be `string | undefined`, why check for `null` if the types say it's impossible?

Reply:
I'm totally happy to change it to `!= null` if that's the style you prefer, I'm just trying to understand the rationale.